src/components/PostCard/PostBodyProposal/PostBodyProposal.js

```diff
@@ -0,0 +1,140 @@
+import cx from 'classnames'
+import React, { useEffect, useState } from 'react'
+import { throttle } from 'lodash/fp'
+import { useTranslation } from 'react-i18next'
+import { useDispatch } from 'react-redux'
+import ReactTooltip from 'react-tooltip'
+import { PROPOSAL_STATUS_CASUAL, PROPOSAL_STATUS_COMPLETED, PROPOSAL_STATUS_DISCUSSION, PROPOSAL_STATUS_VOTING, PROPOSAL_TYPE_MAJORITY, PROPOSAL_TYPE_MULTI_UNRESTRICTED, PROPOSAL_TYPE_SINGLE } from 'store/models/Post'
+import {
+  addProposalVote,
+  removeProposalVote,
+  swapProposalVote
+} from '../../../store/actions/proposals'
+import QuorumBar from 'components/QuorumBar/QuorumBar'
+import './PostBodyProposal.scss'
+import RoundImageRow from 'components/RoundImageRow'
+
+const calcNumberOfVoters = (votes) => {
+  return votes.reduce((acc, vote) => {
+    if (!acc.includes(vote.user.id)) {
+      acc.push(vote.user.id)
+    }
+    return acc
+  }, []).length
+}
+
+const calcNumberOfPossibleVoters = (groups) => {
+  return groups.reduce((acc, group) => {
+    return acc + group.memberCount
+  }, 0)
+}
+
+const isVotingOpen = (proposalStatus) => proposalStatus === PROPOSAL_STATUS_VOTING || proposalStatus === PROPOSAL_STATUS_CASUAL
+
+export default function PostBodyProposal ({
+  currentUser,
+  proposalStatus,
+  proposalType,
+  proposalOptions,
+  proposalVotes,
+  isAnonymousVote,
+  startTime,
+  quorum,
+  endTime,
+  groups,
+  id
+}) {
+  const dispatch = useDispatch()
+  const { t } = useTranslation()
+  const proposalOptionsArray = (proposalOptions && proposalOptions.items) ? proposalOptions?.items : []
+  const proposalVotesArray = (proposalVotes && proposalVotes.items) ? proposalVotes?.items : []
+
+  const [currentUserVotes, setCurrentUserVotes] = useState(proposalVotesArray.filter(vote => vote?.user?.id === currentUser.id))
+  const [currentUserVotesOptionIds, setCurrentUserVotesOptionIds] = useState(currentUserVotes.map(vote => vote.optionId))
+  const [proposalVoterCount, setProposalVoterCount] = useState(calcNumberOfVoters(proposalVotesArray))
```

Comment:
yeah instead of having a useState and a useEffect I think you can just use useMemo for all these calculated values